python/mxnet/symbol/contrib.py

```diff
@@ -336,3 +336,205 @@ def check_data(inputs, in_type, msg):
         states = states[0]
 
     return (outs, states)
+
+def while_loop(cond, func, loop_vars, max_iterations, name="while_loop"):
+    """Run a while loop with user-defined computation and loop condition.
```

Comment:
Is max_iterations always required?

Reply:
yes, without dynamic shape, a user has to provide max_iterations